core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -475,7 +475,52 @@ class DAGSchedulerSuite
     assert(results === Map(0 -> 42, 1 -> 43))
     assertDataStructuresEmpty()
   }
+  
+  test("Test taskAbort after multiple stage failures.") {
```

Comment:
I think this test you wrote has 2 tasks, right?  Because you send two success messages on line 501?